src/mlpack/tests/split_data_test.cpp

```diff
@@ -41,6 +41,22 @@ void CompareData(const mat& inputData,
   }
 }
 
+void CheckMatEqual(const mat& inputData,
+                   const mat& compareData)
+{
+  const mat& sortedInput = arma::sort(inputData, "ascend", 1);
+  const mat& sortedCompare = arma::sort(compareData, "ascend", 1);
+  for (size_t i = 0; i < sortedInput.n_cols; ++i)
+  {
+    const mat& lhsCol = sortedInput.col(i);
+    const mat& rhsCol = sortedCompare.col(i);
+    for (size_t j = 0; j < lhsCol.n_rows; ++j)
+    {
+      BOOST_REQUIRE_CLOSE(lhsCol(j), rhsCol(j), 1e-5);
```

Comment:
This can fail for very small values, I like to use

```
if (std::abs(rhsCol(j)) < 1e-5)
  BOOST_REQUIRE_SMALL(lhsCol(j), 1e-5);
else
  BOOST_REQUIRE_CLOSE(lhsCol(j), rhsCol(j), 1e-5);
```

This is a nice utility function that we could probably use, though.  Actually now that I am thinking about it more... I wonder what would be necessary for us to be able to write `BOOST_REQUIRE_CLOSE(lhs, rhs, 1e-5)` and get the comparison here?  I dunno if that is worth taking the time to figure out though :)